diff --git a/worker.Dockerfile b/worker.Dockerfile
index 39036aa..1f1207a 100644
--- a/worker.Dockerfile
+++ b/worker.Dockerfile
@@ -1,6 +1,6 @@
 FROM rustlang/rust:nightly-bullseye-slim as builder

-RUN apt-get update && apt-get install -y libjemalloc2 libjemalloc-dev make libssl-dev
+RUN apt-get update && apt-get install -y libjemalloc2 libjemalloc-dev make libssl-dev pkg-config

 RUN \
   mkdir -p common/src  && touch common/src/lib.rs && \
@@ -27,7 +27,7 @@ RUN \
   touch ops/src/lib.rs && \
   touch worker/src/main.rs

-RUN cargo build --release --bin worker
+RUN env RUSTFLAGS='-Z linker-features=-lld' cargo build --release --bin worker

 FROM debian:bullseye-slim
 RUN apt-get update && apt-get install -y ca-certificates libjemalloc2
