diff --git a/worker.Dockerfile b/worker.Dockerfile
index ecf7ae9..ada1ee8 100644
--- a/worker.Dockerfile
+++ b/worker.Dockerfile
@@ -1,7 +1,7 @@
 FROM rustlang/rust:nightly-bullseye-slim as builder
 
 # Install jemalloc
-RUN apt-get update && apt-get install -y libjemalloc2 libjemalloc-dev make
+RUN apt-get update && apt-get install -y libjemalloc2 libjemalloc-dev make pkg-config libssl-dev
 
 RUN \
   mkdir -p common/src  && touch common/src/lib.rs && \
