diff --git a/worker.Dockerfile b/worker.Dockerfile
index 39036aa..4e16c45 100644
--- a/worker.Dockerfile
+++ b/worker.Dockerfile
@@ -27,7 +27,7 @@ RUN \
   touch ops/src/lib.rs && \
   touch worker/src/main.rs

-RUN cargo build --release --bin worker
+RUN env RUSTFLAGS='-C target-cpu=native -Z linker-features=-lld' cargo build --release --bin worker

 FROM debian:bullseye-slim
 RUN apt-get update && apt-get install -y ca-certificates libjemalloc2
